plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/GenericStatsMetricPublisher.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.repositories.s3;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+
+/**
+ * Generic stats of repository-s3 plugin.
+ */
+public class GenericStatsMetricPublisher {
```

Comment:
For now it doesn't have any use case to be moved out. If there are other async implementations in other repo types and there is a need of permit based controls then it can be thought of moving out.

Reply:
that makes sense. but till then it might have already been bloated with s3 specific constructs. what do you think @gbbafna ?

Reply:
We can move it later if we have these constructs move out of s3-plugin